### PR TITLE
Device diagnostics instead of compile errors

### PR DESCRIPTION
**Pull request by TwitchBronBron**
Refactor the compileError logic to emit all diagnostics instead of just compile errors. This will allow us to better parse things like device warnings, and also emit component library diagnostics at runtime that don't necessarily crash the app. 

Depends on:
 - #103 
 - https://github.com/rokucommunity/roku-deploy/pull/96

### CODE REVIEW

File: src/CompileErrorProcessor.ts [hunk 1]
```diff
@@ -87,228 +89,238 @@ export class CompileErrorProcessor {
...
+            return x.path && !x.path.toLowerCase().includes('$livecompile');
+        });
+        // return [
+        //     ...this.getSyntaxErrors(this.compilingLines),
+        //     ...this.getCompileErrors(this.compilingLines),
+        //     ...this.getMultipleFileXmlError(this.compilingLines),
+        //     ...this.getSingleFileXmlError(this.compilingLines),
+        //     ...this.getSingleFileXmlComponentError(this.compilingLines),
+        //     ...this.getMissingManifestError(this.compilingLines)
+        // ];
```
Comment (chrisdp) [inline src/CompileErrorProcessor.ts:127]:
```suggestion
```

File: src/CompileErrorProcessor.ts [hunk 2]
```diff
@@ -87,228 +89,238 @@ export class CompileErrorProcessor {
...
-     * @param value
+     * Parse generic xml errors with no further context below
      */
-    private getSyntaxErrorDetails(value: string) {
-        return /(syntax|compile) error.* in (.*)\((\d+)\)(.*)/gim.exec(value);
+    public parseGenericXmlError(line: string): BSDebugDiagnostic[] {
+        // https://regex101.com/r/LDUyww/3
+
+        let [, message, files] = this.execAndTrim(
+            /^(?:-+\>)?\s*(Error parsing (?:multiple )?XML component[s]?)\s+\(?(.+\.xml)\)?.*$/igm,
```
Comment (chrisdp) [inline src/CompileErrorProcessor.ts:137]:
```suggestion
        let [, message, files] = this.execAndTrim(
            // https://regex101.com/r/LDUyww/3
            /^(?:-+\>)?\s*(Error parsing (?:multiple )?XML component[s]?)\s+\(?(.+\.xml)\)?.*$/igm,
```

File: src/CompileErrorProcessor.ts [hunk 3]
```diff
@@ -87,228 +89,238 @@ export class CompileErrorProcessor {
...
-                //FIXME
-                //if this match is a livecompile error, throw out all prior errors because that means we are re-running
-                if (!path.toLowerCase().includes('$livecompile')) {
+    /**
+     * Parse the standard syntax and compile error format
+     */
+    private parseSyntaxAndCompileErrors(line: string): BSDebugDiagnostic[] {
+        // https://regex101.com/r/HHZ6dE/3
+        let [, message, errorType, code, trailingInfo] = this.execAndTrim(
+            /(.*?)(?:\(((?:syntax|compile)\s+error)\s+(&h[\w\d]+)?\s*\))\s*in\b\s+(.+)/ig,
```
Comment (chrisdp) [inline src/CompileErrorProcessor.ts:163]:
```suggestion
        let [, message, errorType, code, trailingInfo] = this.execAndTrim(
            // https://regex101.com/r/HHZ6dE/3
            /(.*?)(?:\(((?:syntax|compile)\s+error)\s+(&h[\w\d]+)?\s*\))\s*in\b\s+(.+)/ig,
```

File: src/CompileErrorProcessor.ts [hunk 4]
```diff
@@ -87,228 +89,238 @@ export class CompileErrorProcessor {
...
+     *  Found 3 parse errors in XML file Foo.xml
+     *  --- Line 2: Unexpected data found inside a <component> element (first 10 characters are "aaa")
+     *  --- Line 3: Some unique error message
+     *  --- Line 5: message with Line 4 inside it
+     */
+    private processMultiLineErrors(lines: string[]): BSDebugDiagnostic[] {
+        const errors = [];
+        // https://regex101.com/r/wBMp8B/1
+        let [, count, filePath] = this.execAndTrim(
+            /found (\d+).*error[s]? in.*?file(.*)/gmi,
```
Comment (chrisdp) [inline src/CompileErrorProcessor.ts:197]:
```suggestion
        let [, count, filePath] = this.execAndTrim(
            // https://regex101.com/r/wBMp8B/1
            /found (\d+).*error[s]? in.*?file(.*)/gmi,
```

File: src/CompileErrorProcessor.ts [hunk 5]
```diff
@@ -87,228 +89,238 @@ export class CompileErrorProcessor {
...
+    /**
+     * Parse error messages that look like this:
+     * ```
+     * ------->No manifest. Invalid package.
+     * ```
+     */
+    private parseMissingManifestError(line: string): BSDebugDiagnostic[] {
+        // https://regex101.com/r/ANr5xd/1
+        let [, message] = this.execAndTrim(
+            /^(?:-+)>(No manifest\. Invalid package\.)/i
```
Comment (chrisdp) [inline src/CompileErrorProcessor.ts:263]:
```suggestion
        let [, message] = this.execAndTrim(
            // https://regex101.com/r/ANr5xd/1
            /^(?:-+)>(No manifest\. Invalid package\.)/i
```

File: src/debugSession/BrightScriptDebugSession.ts [hunk 6]
```diff
@@ -277,24 +277,26 @@ export class BrightScriptDebugSession extends BaseDebugSession {
 
             // handle any compile errors
             // eslint-disable-next-line @typescript-eslint/no-misused-promises
-            this.rokuAdapter.on('compile-errors', async (errors: BrightScriptDebugCompileError[]) => {
+            this.rokuAdapter.on('diagnostics', async (errors: BSDebugDiagnostic[]) => {
                 // remove redundant errors and adjust the line number:
                 // - Roku device and sourcemap work with 1-based line numbers,
                 // - VS expects 0-based lines.
                 const compileErrors = util.filterGenericErrors(errors);
```
Comment (chrisdp) [inline src/debugSession/BrightScriptDebugSession.ts:284]: why?

Comment (TwitchBronBron) [reply src/debugSession/BrightScriptDebugSession.ts]: Yeah...that can go bye bye

**Pull request merged**